client/client_test.go

```diff
@@ -383,63 +383,75 @@ func TestLoadingTrust(t *testing.T) {
+	intvl, err := GetExponentialBackoffTime(0, 1*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(1, 1*time.Minute)
+	intvl, err = GetExponentialBackoffTime(1, 1*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(2, 1*time.Minute)
+	intvl, err = GetExponentialBackoffTime(2, 1*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	_, err = GetExponentialBackoffTime(3, 1*time.Minute)
+	_, err = GetExponentialBackoffTime(3, 1*time.Minute, 0)
 	assert.Error(t, err)
 
-	_, err = GetExponentialBackoffTime(7, 1*time.Minute)
+	_, err = GetExponentialBackoffTime(7, 1*time.Minute, 0)
 	assert.Error(t, err)
 
 	// Test with two minute maximum interval.
-	intvl, err = GetExponentialBackoffTime(5, 2*time.Minute)
+	intvl, err = GetExponentialBackoffTime(5, 2*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 2*time.Minute)
 
-	_, err = GetExponentialBackoffTime(6, 2*time.Minute)
+	_, err = GetExponentialBackoffTime(6, 2*time.Minute, 0)
 	assert.Error(t, err)
 
 	// Test with 10 minute maximum interval.
-	intvl, err = GetExponentialBackoffTime(11, 10*time.Minute)
+	intvl, err = GetExponentialBackoffTime(11, 10*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 8*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(12, 10*time.Minute)
+	intvl, err = GetExponentialBackoffTime(12, 10*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 10*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(14, 10*time.Minute)
+	intvl, err = GetExponentialBackoffTime(14, 10*time.Minute, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 10*time.Minute)
 
-	_, err = GetExponentialBackoffTime(15, 10*time.Minute)
+	_, err = GetExponentialBackoffTime(15, 10*time.Minute, 0)
 	assert.Error(t, err)
 
 	// Test with one second maximum interval.
-	intvl, err = GetExponentialBackoffTime(0, 1*time.Second)
+	intvl, err = GetExponentialBackoffTime(0, 1*time.Second, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(1, 1*time.Second)
+	intvl, err = GetExponentialBackoffTime(1, 1*time.Second, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	intvl, err = GetExponentialBackoffTime(2, 1*time.Second)
+	intvl, err = GetExponentialBackoffTime(2, 1*time.Second, 0)
 	assert.NoError(t, err)
 	assert.Equal(t, intvl, 1*time.Minute)
 
-	_, err = GetExponentialBackoffTime(3, 1*time.Second)
+	_, err = GetExponentialBackoffTime(3, 1*time.Second, 0)
 	assert.Error(t, err)
+
+	maxAttempts := 8
+	expectedIntervalMinutes := 2
+	for try := 1; try < maxAttempts; try++ {
+		intvl, err := GetExponentialBackoffTime(try, 1*time.Minute, maxAttempts)
+		assert.NoError(t, err)
+		assert.Equal(t, time.Duration(expectedIntervalMinutes)*time.Minute, intvl)
+		expectedIntervalMinutes *= 2
```

Comment:
This will need to be adjusted as well. I would not repeat the formula here, but rather test exactly the values we expect at various extremes.

Reply:
ok.